katsdptelstate/aio/redis.py

```diff
@@ -289,12 +299,14 @@ async def _run_pubsub(self) -> None:
         """
         try:
             loop = asyncio.get_event_loop()
+            # Ensure we are always subscribed to something, as a workaround for
```

Comment:
So this is only on the connection dealing with subscriptions?

Reply:
There is also the synchronous telstate redis backend. In theory that might also be vulnerable to https://github.com/redis/redis-py/issues/1720, but it uses a dedicated connection for each call to `wait_keys` and hence shouldn't run into problems unless something stalls one of the initial subscriptions for more than 30 seconds - highly unlikely and would probably cause the socket timeout to expire anyway. The aioredis backend was changed to multiplex over a single pub-sub connection because something (I think ingest) needed to monitor a large number of keys and having a connection per wait_key was running into some limit.